packages/agents-hosting/src/app/teamsAttachmentDownloader.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Attachment, Channels } from '@microsoft/agents-activity'
+import { debug } from '@microsoft/agents-activity/logger'
+import { ConnectorClient } from '../connector-client'
+import { InputFile, InputFileDownloader } from './inputFileDownloader'
+import { TurnContext } from '../turnContext'
+import { TurnState } from './turnState'
+import axios, { AxiosInstance } from 'axios'
+import { z } from 'zod'
+
+const logger = debug('agents:teamsAttachmentDownloader')
+
+/**
+ * Downloads attachments from Teams using the bots access token.
+ */
+export class TeamsAttachmentDownloader<TState extends TurnState = TurnState> implements InputFileDownloader<TState> {
+  private _httpClient: AxiosInstance
+  private _stateKey: string
+
+  public constructor (stateKey: string = 'inputFiles') {
+    this._httpClient = axios.create()
+    this._stateKey = stateKey
+  }
+
+  /**
+   * Download any files relative to the current user's input.
+   *
+   * @param {TurnContext} context Context for the current turn of conversation.
+   * @returns {Promise<InputFile[]>} Promise that resolves to an array of downloaded input files.
+   */
+  public async downloadFiles (context: TurnContext): Promise<InputFile[]> {
+    if (context.activity.channelId !== Channels.Msteams && context.activity.channelId !== Channels.M365Copilot) {
+      return Promise.resolve([])
+    }
+    // Filter out HTML attachments
+    const attachments = context.activity.attachments?.filter((a) => a.contentType && !a.contentType.startsWith('text/html'))
+    if (!attachments || attachments.length === 0) {
+      return Promise.resolve([])
+    }
+
+    const connectorClient : ConnectorClient = context.turnState.get<ConnectorClient>(context.adapter.ConnectorClientKey)
+    this._httpClient.defaults.headers = connectorClient.axiosInstance.defaults.headers
+
+    const files: InputFile[] = []
+    for (const attachment of attachments) {
+      const file = await this.downloadFile(attachment)
+      if (file) {
+        files.push(file)
+      }
+    }
+
+    return files
+  }
+
+  /**
+   * @private
+   * @param {Attachment} attachment - Attachment to download.
+   * @returns {Promise<InputFile>} - Promise that resolves to the downloaded input file.
+   */
+  private async downloadFile (attachment: Attachment): Promise<InputFile | undefined> {
+    let inputFile: InputFile | undefined
+
+    if (attachment.contentUrl && (attachment.contentUrl.startsWith('https://') || attachment.contentUrl.startsWith('http://localhost'))) {
```

Comment:
when would it ever be http://localhost?

Reply:
Maybe @tracyboehrer can help us with that.

Reply:
Teams has a couple different ways.  I would highly recommend what the DotNet version is doing and roll with that.  It doesn't have to be an exact line-by-line match, but in spirit.